lib/controller.js

```diff
@@ -641,6 +663,213 @@ controller.tesselFirmwareVerion = function(opts) {
+  },
+  exit: function(resolve) {
+    resolve();
+    //process.exit();
+  }
+};
+controller.ssh = {
+  runSSH: function(id, opts, tessels, resolve, reject) {
+    // clear console
+    debug('runSSH...\n root@' + tessels[id].lanConnection.host);
+    if (opts.menu) {
+      // because the test needs to resolve the promise ...
+      resolve({
+        opts: opts,
+        tessels: tessels
+      });
+    } else {
+      // if this is no test, starting the menu as child process
+      controller.menu.clear();
+      var ch = require('child_process')
+        .spawn('ssh', ['-i',
+          opts.path,
+          'root@' + tessels[id].lanConnection.ip
+        ], {
+          stdio: 'inherit'
+        });
+      logs.info('Connect to ' + tessels[id].lanConnection.host + '...');
+
+      // FIXME: There is no handler for poweroff Tessel ! Needs to be fixed within the firmware...
+      // (Terminal freezes imediatelly after poweroff the Tessel)
+
+      ch.once('error', function(e) {
+        if (e === 255) {
+          controller.menu.clear();
+          controller.ssh.notAuthorized();
+          reject();
+        } else {
+          logs.warn('Error while connected to ' + tessels[id].lanConnection.host + ':', e);
+          reject(e);
+        }
+
+      });
+      ch.once('exit', function(e) {
+
+        if (e === 255) {
+          controller.menu.clear();
+          controller.ssh.notAuthorized();
+          reject();
+        } else if (e === 127) { // exit by user
+          // clear console
+          controller.menu.clear();
+          logs.warn('Connection to ' + tessels[id].lanConnection.host + ' closed!\n');
+          resolve();
+        } else if (e === 0) {
+          // everything works fine ... now lets clear the terminal
+          controller.menu.clear();
+          logs.info('Connection to ' + tessels[id].lanConnection.host + ' closed!\n');
+          resolve();
+        } else {
+          logs.warn('Connection to ' + tessels[id].lanConnection.host + ' closed due to reason:\n', e);
+          reject(e);
+        }
+      });
+
+      ch.once('close', function(e) {
+        // tessel powers off
+        if (e === 0) {
+          // everything works fine...
+          resolve();
+        } else if (e === 255) {
+          //reject();
+          process.exit();
+        } else if (e === 127) {
+          reject();
+        } else {
+          logs.warn('Connection to ' + tessels[id].lanConnection.host + ' closed due to reason:\n', e);
+          reject();
+        }
+      });
+    }
+  },
+  notAuthorized: function() {
+    logs.warn('Sorry, you are not authorized!');
+    logs.info('Maybe this Tessel didn\'t got "t2 provision" until now?');
+  }
+};
+
+/*
+  The T2 root command is used to login into the firmware and gaining superuser access.
+  The security is provided by RSA - your Tessel get the keys while provisioning via USB.
+  
+  If you have only one Tessel, the T2 root command will login directly else there is 
```

Comment:
Yep! Just like if you only have one Tessel and you do `t2 run`, it will automatically deploy to that Tessel. It provides for the best user experience because most people will only have one Tessel.